rasa/core/nlg/response.py

```diff
@@ -34,7 +34,11 @@ def _matches_filled_slots(
         for constraint in constraints:
             name = constraint["name"]
             value = constraint["value"]
-            if filled_slots.get(name) != value:
+            filled_slots_value = filled_slots.get(name)
+            if isinstance(filled_slots_value, str) and isinstance(value, str):
+                if filled_slots_value.casefold() == value.casefold():
```

Comment:
Optional (nitpicking): To avoid nested ifs for better readability, I'd extract this condition into a bool variable:
```suggestion
            is_string_check = isinstance(filled_slots_value, str) and isinstance(value, str)
            if is_string_check and (filled_slots_value.casefold() == value.casefold()):
```

Reply:
Thank you for the suggestion, I have changed the code accordingly